habitat_baselines/utils/common.py

```diff
@@ -171,21 +173,42 @@ def batch_obs(
 
     for sensor_name in sensor_names:
         for i, obs in enumerate(observations):
-            sensor = torch.as_tensor(obs[sensor_name])
+            sensor = obs[sensor_name]
             if cache is None:
-                batch[sensor_name].append(sensor)
+                batch[sensor_name].append(torch.as_tensor(sensor))
             else:
                 if sensor_name not in batch_t:
                     batch_t[sensor_name] = cache.get(
-                        len(observations), sensor_name, sensor, device
+                        len(observations),
+                        sensor_name,
+                        torch.as_tensor(sensor),
+                        device,
                     )
 
-                batch_t[sensor_name][i].copy_(sensor)
+                # Use isinstance(sensor, np.ndarray) here instead of
+                # np.asarray as this is quickier for the more common
+                # path of sensor being an np.ndarray
+                # np.asarray is ~3x slower than checking
+                if isinstance(sensor, np.ndarray):
```

Comment:
@erikwijmans I think I found a better solution here: np.asanyarray https://stackoverflow.com/a/56805220/2444240

Reply:
That's fixing a different problem, `np.asanyarray` slower than `np.asarray` by a very small but still measurable amount in my testing.

Reply:
Still I think we should replace the asarray at the bottom with asanyarray.

Reply:
Why?  asanyarray just lets subclasses of `np.ndarray` pass through, but we know sensor cannot be a subclass of `np.ndarray` if that if `isinstance(sensor, np.ndarray)` returned false.